### PR TITLE
Upgrades to use .NET 8 RC1

### LINKED ISSUE #3678: Upgrade to .NET 8
We want to upgrade the code to build only with .NET 8, and remove a few .NET 7 work-arounds in the source code.

### PR DESCRIPTION
**Pull request by pepone**
Fix #3678

### CODE REVIEW

File: src/IceRpc/Internal/ProtocolLoggerExtensions.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ internal static partial class ProtocolLoggerExtensions
             remoteNetworkAddress);
 
     // Multiple logging methods are using same event id.
-#pragma warning disable SYSLIB1006
+#pragma warning disable SYSLIB1025
```
Comment (externl) [inline src/IceRpc/Internal/ProtocolLoggerExtensions.cs:65]: The number was changed?

Comment (pepone) [reply src/IceRpc/Internal/ProtocolLoggerExtensions.cs:65]: Yes I get this new error with Visual Studio 2022 preview

**Review by externl — APPROVED** [review]

File: src/IceRpc/Internal/ProtocolLoggerExtensions.cs (hunk 1 above)
Comment (bernardnormier) [reply src/IceRpc/Internal/ProtocolLoggerExtensions.cs:65]: I think we should use 2 separate EventId here - for another PR!

**Review by bernardnormier — APPROVED** [review]
Looks good.

**Pull request merged**